smslibrary/src/main/java/com/eis/smslibrary/SMSPeer.java

```diff
@@ -4,56 +4,125 @@
 
 import com.eis.communication.Peer;
 import com.eis.smslibrary.exceptions.InvalidTelephoneNumberException;
+import com.google.i18n.phonenumbers.NumberParseException;
+import com.google.i18n.phonenumbers.PhoneNumberUtil;
+import com.google.i18n.phonenumbers.Phonenumber;
+
+import java.util.Arrays;
+import java.util.Locale;
 
 /**
  * Implementation of {@link Peer} for SMS communication channel.
  *
  * @author Luca Crema, Marco Mariotto. Reviewed by Marco Cognolato and Mattia Fanan
+ * @author Matteo Carnelos
  */
-public class SMSPeer implements Peer<String>, Comparable<SMSPeer> {
+public class SMSPeer implements Peer<String>, Comparable<SMSPeer>, java.io.Serializable {
+
+    private static final String REGION_NEEDED_EXC_MSG = "The given number does not explicitly " +
+            "contain the prefix and the default region is not obtainable. Please use the method " +
+            "SMSPeer.setDefaultRegion() in order to set a valid one.";
+    private static final String INVALID_NUMBER_EXC_MSG = "Invalid telephone number: '%s'. Note that " +
+            "a telephone number pointing outside the currently set default region (%s) should " +
+            "be passed with also the prefix or it might be considered invalid.";
+    private static final String NOT_PARSABLE_EXC_MSG = "Not parsable telephone number: '%s'. Please " +
```

Comment:
I don't like how you handled the errors with strings and not with an enum because what if I were to translate the app in Italian? I couldn't even check if `exception.message.equals(SMSPeer.REGION_NEEDED_EXC_MSG)` because they're private constants.
I would have used an enum of possible errors like it used to be and let the user write the message for each error.

Reply:
That's legit. You're right. I'll restore the struct with the new types of error: `REGION_NEEDED`, `INVALID_NUMBER`, `NOT_PARSABLE_NUMBER`.